### ISSUE TITLE
[Feature Request] Add support for .swift-version file and older swift versions

### ISSUE THREAD

**Issue by Cyberbeni** [realm/SwiftLint#3934]
### New Issue Checklist

- [x] Updated SwiftLint to the latest version
- [x] I searched for [existing GitHub issues](https://github.com/realm/SwiftLint/issues)

### Description

We are running UI tests for our macOS app starting with macOS 10.11, meaning that the test runner has to support Swift 3.0 (The linting is always done on the latest OS with the latest Xcode)

SwiftFormat allows us to have a .swift-version file in random folders that override the swift version specified in the main config file, so we don't have to bother with manually configuring which rules are enabled where.

I noticed this because SwiftLint started complaining about the legacy_random rule which has been enabled here, along with removing the minimum swift version requirement for a bunch of rules that were below 5.0: https://github.com/realm/SwiftLint/pull/3886/files

##### Complete output when running SwiftLint, including the stack trace and command used

```bash
ln -s "$GIT_ROOT_DIR/.swiftlint.yml" "$SRCROOT/.swiftlint.yml"
"$($BREW --prefix)/bin/swiftlint" --path "$SRCROOT"
```

The .swift-version file containing `3.0` is also in SRCROOT

```console
...
/path/to/file.swift:28:33: warning: Legacy Random Violation: Prefer using `type.random(in:)` over legacy functions. (legacy_random)
...
```

### Example code

```swift
import Foundation

#if swift(>=4.2)

#else
    public extension Array {
        func randomElement() -> Element? {
            guard !isEmpty else { return nil }

            return self[Int(arc4random_uniform(UInt32(count)))]
        }

        func allSatisfy(_ predicate: (Element) throws -> Bool) rethrows -> Bool {
            return try !contains {
                return try !predicate($0)
            }
        }
    }

#endif
```

**Comment by jpsim** [comment]
As of SwiftLint 0.47.0, SwiftLint should only be used to lint projects using Swift 5.0 or later, as per this changelog entry in the "Breaking" section:

> SwiftLint now requires at least Swift 5.0 installed in order to lint files.

You can certainly use older versions of SwiftLint prior to 0.47.0 to keep linting projects that use an older version of Swift.

**Comment by jpsim** [comment]
SwiftLint does not differentiate this today. The version of Swift that you get from `xcrun swift --version` is the version of Swift it will assume your files are built with.

I'm happy to keep this feature request open if anyone wants to come up with a comprehensive design for how this should work.
